### DISCUSSION TITLE
Validating that a field is unique OR the same value it was before

### DISCUSSION THREAD

**Discussion by parker-codes** [adonisjs/core#1513] [General]
It wasn't documented for this use case so I hope this helps someone. 

## Use Case

A user can visit a profile page to update their details (name, email, and phone number). If they change their email, the **unique** validation rule passes. If they leave it the same, however (which is usually the case), it will fail.

## Solution

I had to use the `whereNot` property and provide it the existing email value, like so:

```js
email: schema.string({ trim: true }, [
  rules.email(),
  rules.unique({
    table: 'users',
    column: 'email',
    // this allows the email to be the same as it was before
    whereNot: { email: user.email },
  }),
]),
```

(note that [the docs](https://preview.adonisjs.com/guides/validator/rules#additional-constraints) mention using refs if you are using schema caching)

**Comment by amirhoseinsalimi** [comment]
Wow... saved me today!
